common/src/sql/dbinit.sql

```diff
@@ -669,15 +665,15 @@ CREATE TABLE omicron.public.identity_provider (
     provider_type omicron.public.provider_type NOT NULL
 );
 
-CREATE INDEX ON omicron.public.identity_provider (
-    id,
-    silo_id
+CREATE UNIQUE INDEX ON omicron.public.identity_provider (
+    silo_id,
+    id
 ) WHERE
     time_deleted IS NULL;
 
-CREATE INDEX ON omicron.public.identity_provider (
-    name,
-    silo_id
+CREATE UNIQUE INDEX ON omicron.public.identity_provider (
+    silo_id,
+    name
```

Comment:
**This is one of the few functional changes in this PR.**  This index was _not_ unique and I think that was a bug.  See [this comment](https://github.com/oxidecomputer/omicron/issues/1497#issuecomment-1464072388).

I also flipped the order of fields in order to use this index to paginate the IdPs in a Silo by name.